doc/sphinxext/gallery_generator.py

```diff
@@ -240,9 +240,13 @@ def thumbfilename(self):
 
     @property
     def apiname(self):
-        name = self.modulename.split("_")
-        name = name[1::]
-        return "_".join(name)
+        name=""
+        with open(op.join(self.target_dir, self.pyfilename), "r") as file:
+            regex = r"az\.(plot\_[a-z_]+)\("
+            matches = re.finditer(regex, file.read(), re.MULTILINE)
+            for matchNum, match in enumerate(matches, start=1):
+                name = match.group(1)
```

Comment:
will this return the last found `plot_xyz`? Should it return all the matches?

Do you think `re.findall` would work here? No need for looping?

Reply:
I think the first is enough, there is only a couple of cases where more than a function is called, and in these cases the function called is always the same.

Reply:
@ahartikainen @OriolAbril I worked some examples and let's say we have a hypothetical code as:
```
az.plot_compare(model_compare, figsize=(12, 4))
az.plot_forest(model_compare, figsize=(12, 4))
az.plot_density(model_compare, figsize=(12, 4))
```
If we run our code, it will return the last found match as 'plot_density'.  As in our example files, we are only calling the function one time, so I don't think it'll be a problem.
As a possible solution for, when (in future) we have multiple plots in a single example, we can do something like:
```
    for matchNum, match in enumerate(matches, start=1):
        api_name.append(match.group(1))
    return api_name #returns the list
```
And then maybe check the length when rendering it in the template.
As suggested by @ahartikainen , `re.findall()` will work best for such case. For example, above code can be simplified as
```
with open("mpl_plot_compare.py", "r") as f:
    regex = r"az\.(plot\_[a-z_]+)\("
    api_name = re.findall(regex, f.read())
return api_name  #api_name will contain list
```
How should we do it?

Reply:
I think using only the first function is good enough for that. Returning the list with all occurrences seems like an overkill which complicates the code without much gain